diff --git a/src/ModelingData/TKGeomBase/GeomTools/GeomTools_Curve2dSet.cxx b/src/ModelingData/TKGeomBase/GeomTools/GeomTools_Curve2dSet.cxx
index 5133b9d..36f8f99 100644
--- a/src/ModelingData/TKGeomBase/GeomTools/GeomTools_Curve2dSet.cxx
+++ b/src/ModelingData/TKGeomBase/GeomTools/GeomTools_Curve2dSet.cxx
@@ -68,7 +68,7 @@ void GeomTools_Curve2dSet::Clear()
 
 int GeomTools_Curve2dSet::Add(const occ::handle<Geom2d_Curve>& S)
 {
-  return myMap.Add(S);
+  return (S.IsNull()) ? 0 : myMap.Add(S);
 }
 
 //=================================================================================================
@@ -87,7 +87,7 @@ occ::handle<Geom2d_Curve> GeomTools_Curve2dSet::Curve2d(const int I) const
 
 int GeomTools_Curve2dSet::Index(const occ::handle<Geom2d_Curve>& S) const
 {
-  return myMap.FindIndex(S);
+  return (S.IsNull()) ? 0 : myMap.FindIndex(S);
 }
 
 //=================================================================================================
diff --git a/src/ModelingData/TKGeomBase/GeomTools/GeomTools_SurfaceSet.cxx b/src/ModelingData/TKGeomBase/GeomTools/GeomTools_SurfaceSet.cxx
index 32364cd..001e351 100644
--- a/src/ModelingData/TKGeomBase/GeomTools/GeomTools_SurfaceSet.cxx
+++ b/src/ModelingData/TKGeomBase/GeomTools/GeomTools_SurfaceSet.cxx
@@ -73,7 +73,7 @@ void GeomTools_SurfaceSet::Clear()
 
 int GeomTools_SurfaceSet::Add(const occ::handle<Geom_Surface>& S)
 {
-  return myMap.Add(S);
+  return (S.IsNull()) ? 0 : myMap.Add(S);
 }
 
 //=================================================================================================
@@ -91,7 +91,7 @@ occ::handle<Geom_Surface> GeomTools_SurfaceSet::Surface(const int I) const
 
 int GeomTools_SurfaceSet::Index(const occ::handle<Geom_Surface>& S) const
 {
-  return myMap.FindIndex(S);
+  return (S.IsNull()) ? 0 : myMap.FindIndex(S);
 }
 
 //=================================================================================================
