diff --git a/src/ModelingAlgorithms/TKGeomAlgo/GeomFill/GeomFill_CoonsAlgPatch.cxx b/src/ModelingAlgorithms/TKGeomAlgo/GeomFill/GeomFill_CoonsAlgPatch.cxx
index e3abbaab..f0c1fd21 100644
--- a/src/ModelingAlgorithms/TKGeomAlgo/GeomFill/GeomFill_CoonsAlgPatch.cxx
+++ b/src/ModelingAlgorithms/TKGeomAlgo/GeomFill/GeomFill_CoonsAlgPatch.cxx
@@ -91,14 +91,14 @@ gp_Pnt GeomFill_CoonsAlgPatch::Value(const double U, const double V) const
   a3 = 1. - a1;
   gp_XYZ cor, cortemp;
 
-  cor = bound[0]->Value(V).XYZ();
+  cor = bound[0]->Value(U).XYZ();
   cor.Multiply(a0);
 
   cortemp = bound[1]->Value(V).XYZ();
   cortemp.Multiply(a1);
   cor.Add(cortemp);
 
-  cortemp = bound[2]->Value(V).XYZ();
+  cortemp = bound[2]->Value(U).XYZ();
   cortemp.Multiply(a2);
   cor.Add(cortemp);
 
